crates/full-node/sov-db/src/flat_db.rs

```diff
@@ -203,10 +208,14 @@ impl FlatStateDb {
             &self.live_db,
         )?;
 
-        // Write batches.
+        // Write archival db batches.
+        commit_flag.save_commit_status(&CommitStatus::CommittingArchivalUserAndKernel)?;
```

Comment:
Nit: what do you think about debug_assert that previous state is matching expected state of NOMT being commited (without actually checking root hash)

Reply:
Inside `save_commit_status` there is a check that the status was saved. 
```
   pub(crate) fn save_commit_status(&self, commit_status: &CommitStatus) -> anyhow::Result<()> {
        self.write_status(commit_status).with_context(|| {
            format!("Failed to write {commit_status:?} status after successful user commit")
        })?;

        debug_assert_eq!(&self.read_status()?, commit_status);
        Ok(())
    }
```